components/notebook-controller/pkg/controller/notebook/notebook_controller.go

```diff
@@ -123,18 +122,81 @@ func (r *ReconcileNotebook) Reconcile(request reconcile.Request) (reconcile.Resu
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
-	if err = r.ReconcileStatefulSet(instance); err != nil {
+
+	// Reconcile StatefulSet
+	ss := generateStatefulSet(instance)
+	if err := controllerutil.SetControllerReference(instance, ss, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
-	if err = r.ReconcileService(instance); err != nil {
+	// Check if the StatefulSet already exists
+	foundStateful := &appsv1.StatefulSet{}
+	justCreated := false
+	err = r.Get(context.TODO(), types.NamespacedName{Name: ss.Name, Namespace: ss.Namespace}, foundStateful)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Creating StatefulSet", "namespace", ss.Namespace, "name", ss.Name)
+		err = r.Create(context.TODO(), ss)
+		justCreated = true
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else if err != nil {
 		return reconcile.Result{}, err
 	}
+	// Update the foundStateful object and write the result back if there are any changes
+	if !justCreated && util.CopyStatefulSetFields(ss, foundStateful) {
```

Comment:
What if StatefulSet with name ```ss.Name``` already exist in the namespace and is created by other user / service?

Reply:
I think the reconcile will then fail, which is the intended behavior?

Reply:
Ideally we should fail it and add message to status explaining why it fails.
Seems currently controller will just update StatefulSet?

Reply:
I think it will fail because we can't claim ownership on that statefulset.